app/controllers/imports_controller.rb

```diff
@@ -47,6 +48,24 @@ def load_csv
     end
   end
 
+  def upload_csv
+    raw_csv = params[:file].read
+
+    begin
+      csv = Import::Csv.new(raw_csv)
+      if csv.valid?
+        render turbo_stream: turbo_stream.action(
+          :update_input, "raw_csv_text_area", raw_csv
+        )
+      else
+        render json: { message: "CSV contents is not valid" }, status: :unprocessable_entity
```

Comment:
The errors are not translated, since they are only logged in the console here, which could be picked up by a service like Sentry.

The dropzone does translate errors as seen in the translation file.

Reply:
No worries, think this is okay!